"Add STML help button widget with basic UI and tests" without lastest errors and updated with return me home function

Layesuza opened the pull request:
## Description
This PR reintroduces the emergency management feature to avoid branch conflicts from the previous PR. I have reused the same files from the old PR and created a new PR to ensure a clean merge.

## Changes
- Reused the following files from the previous PR:
  - `lib/models/emergency_type.dart`
  - `lib/main.dart`
  - `lib/models/location_entry.dart`
  - `lib/services/caregiver_notification_service.dart`
  - `lib/services/emergency_service.dart`
  - `lib/services/location_database.dart`
  - `lib/services/navigation_service.dart`
  - `lib/ui/home_screen.dart`
  - `lib/viewmodels/emergency_help_viewmodel.dart`
  - `lib/widgets/emergency_help_button.dart`

## Context
- The previous PR had branch conflicts, so I decided to create a new PR to avoid these issues.
- All old branches related to this feature have been deleted to keep the repository clean.

## Testing
- Ensure all changes are consistent with the previous implementation.
- Verify that the emergency management feature works as expected.

## Notes for Reviewers
- This PR contains the same changes as the previous one but is based on the latest version of the main branch to avoid conflicts.
- Please review the changes and let me know if any adjustments are needed.

Thank you for your time and feedback! 🙏




updated :
## Emergency Features Implementation  
**Changes:**  
- ✅ "Return Me Home" GPS navigation  
- ✅ Twilio SMS integration for 911 calls  
- ✅ Firebase Cloud Messaging for caregiver alerts  
- 🛠️ Backend API endpoints for emergency workflows  

**Testing Steps:**  
1. Simulate an emergency call via the dedicated button
2. Verify the reception of SMS and notifications
3. Validate the synchronization of contacts

create mode 100644 STML/stml_application/.firebaserc
 create mode 100644 STML/stml_application/functions/.eslintrc.js
 create mode 100644 STML/stml_application/functions/.gitignore
 create mode 100644 STML/stml_application/functions/index.js
 create mode 100644 STML/stml_application/functions/package-lock.json
 create mode 100644 STML/stml_application/functions/package.json
 create mode 100644 STML/stml_application/lib/config/config.dart
 delete mode 100644 STML/stml_application/lib/main.dart
 create mode 100644 STML/stml_application/lib/services/esri_client.dart
 create mode 100644 STML/stml_application/lib/ui/caregiver_settings_screen.dart
 create mode 100644 STML/stml_application/test/services/emergency_service_test.dart
 create mode 100644 STML/stml_application/test/viewmodels/emergency_help_viewm

Layesuza committed:
"Add STML help button widget with basic UI and tests"

Layesuza committed:
Merge branch 'developer' of https://github.com/umgc/spring2025 into developer

Layesuza committed:
Merge remote-tracking branch 'origin/developer' into feature/help-button

Layesuza committed:
git push origin main

Layesuza committed:
Resolved merge conflicts

Layesuza committed:
Merge branch 'developer' of https://github.com/umgc/spring2025 into developer

aholt014 reviewed (approved):
rebase and resolve any conflicts before merging

aholt014 commented:
NOTE: just add the new files you created do not merge the entire project if the rebase tells you to do. just resolve the conflicts

aholt014 commented on STML/stml_application/test/ui/widgets/emergency_help_button_test.dart:
this does not work when ran

Layesuza committed:
Add emergency management feature

- Added EmergencyType model to define emergency categories
- Implemented LocationEntry model for tracking user locations
- Integrated CaregiverNotificationService for real-time alerts
- Developed EmergencyService to handle emergency requests
- Added LocationDatabase for persistent location storage
- Implemented NavigationService for route guidance
- Created HomeScreen UI for user interaction
- Added EmergencyHelpViewModel to manage emergency logic
- Designed EmergencyHelpButton widget for quick access

This commit introduces the core functionality for managing emergencies, including models, services, and UI components.

ashekh9 commented:
I am getting a bunch of errors in home_screen.dart with undefined parameters, nonexistent classes, missing classes, and a bunch of import errors for these files:
import 'package:memoryminder/ui/assistant_screen.dart';
import 'package:memoryminder/ui/audio_screen.dart';
import 'package:memoryminder/ui/gallery_screen.dart';
import 'package:memoryminder/ui/location_history_screen.dart';
import 'package:memoryminder/ui/profile_screen.dart';
import 'package:memoryminder/ui/response_screen.dart';
import 'package:memoryminder/ui/tour_screen.dart';
import 'package:memoryminder/src/camera_manager.dart';
import 'package:memoryminder/src/utils/ui_utils.dart';

Layesuza committed:
Ignore team_b change .gitignore

Layesuza committed:
Ignorer team_b dans .gitignore

Layesuza committed:
Merge branch 'developer' into feature/help-button

This commit merges the latest changes from the 'developer' branch into 'feature/help-button'.
- Resolved merge conflicts in [list affected files, if applicable].
- Integrated new features/bug fixes from 'developer'.

Layesuza committed:
feat: add emergency features and backend configuration

- Implement 'Return Me Home' navigation functionality
- Configure backend API endpoints (POST /emergency, /location, /notify)
- Add 911 emergency call feature with Twilio SMS integration
- Implement caregiver notification system via Firebase Cloud Messaging
- Update emergency contact synchronization logic

Layesuza commented:
> I am getting a bunch of errors in home_screen.dart with undefined parameters, nonexistent classes, missing classes, and a bunch of import errors for these files: import 'package:memoryminder/ui/assistant_screen.dart'; import 'package:memoryminder/ui/audio_screen.dart'; import 'package:memoryminder/ui/gallery_screen.dart'; import 'package:memoryminder/ui/location_history_screen.dart'; import 'package:memoryminder/ui/profile_screen.dart'; import 'package:memoryminder/ui/response_screen.dart'; import 'package:memoryminder/ui/tour_screen.dart'; import 'package:memoryminder/src/camera_manager.dart'; import 'package:memoryminder/src/utils/ui_utils.dart';

i've allready fix it and updated the code.... please re-check it

aholt014 commented:
i am rejecting this PR and creating a new branch and will cherrypick the changes you have to eliminate more confusion

aholt014 commented:
revise this PR, test to see if it actually does what you want it to do. please include screenshots if you are able

also, do not delete the existing main.dart file. that is the entry point of the app that all current code needs and your main.dart function doesn't have 1/2 of what the current main.dart has so im pretty sure if pushed it will break things lol

again you will need to **_rebase_** instead of merge to get the recent changes on the developer branch and then accept the changes you need, hence the conflicts

aholt014 reviewed (changes requested):
needs work. all 340 files do not need to be merged. also my brain is on turtle atm but i think you will be deleting team a and b work w/ what your currently have?? i could be wrong tho

use the following as guidance on how to remove the unnecessary files: https://graphite.dev/guides/how-to-remove-a-file-from-a-git-commit
